llvm/utils/TableGen/CodeEmitterGen.cpp

```diff
@@ -448,15 +452,13 @@ void CodeEmitterGen::emitInstructionBaseValues(
 void CodeEmitterGen::emitCaseMap(
     raw_ostream &o,
     const std::map<std::string, std::vector<std::string>> &CaseMap) {
-  std::map<std::string, std::vector<std::string>>::const_iterator IE, EE;
-  for (IE = CaseMap.begin(), EE = CaseMap.end(); IE != EE; ++IE) {
-    const std::string &Case = IE->first;
-    const std::vector<std::string> &InstList = IE->second;
-
-    for (int i = 0, N = InstList.size(); i < N; i++) {
-      if (i)
+  for (const auto &[Case, InstList] : CaseMap) {
+    bool First = false;
```

Comment:
Shouldn't this be `true`?

Reply:
Yes, thanks for catching it. I think we should change such code to use the `ListSeparator` API that avoids such problems. But I did not want to introduce that in this change.

Reply:
Fixed now.